ロールタイムラインタブを追加できるように #327

tacko-o opened the pull request:
タブ設定からロールタイムラインを追加できるようにしました
[#327 ロールタイムラインを取得したい](https://github.com/shiosyakeyakini-info/miria/issues/327)
![roleTimeline](https://github.com/shiosyakeyakini-info/miria/assets/140358540/6f93823b-0716-4d18-8434-8e0e17b017d1)

https://github.com/shiosyakeyakini-info/misskey_dart/pull/24 に依存します

tacko-o committed:
ロールタイムラインタブを追加できるように

タブ設定でロールタイムラインを指定できるようにしました。

poppingmoon commented on lib/repository/account_repository.dart:
```diff
@@ -107,6 +107,11 @@ class AccountRepository extends ChangeNotifier {
             tabType: TabType.globalTimeline,
             name: "グローバルタイムライン",
             account: account),
+        TabSetting(
+            icon: TabIcon(codePoint: Icons.bookmark.codePoint),
+            tabType: TabType.roleTimeline,
+            name: "ロールタイムライン",
+            account: account),
```
```suggestion
```

ここでは一つ目のアカウントが追加されたときに自動で追加されるタブが記述されています
ロールタイムラインは利用頻度が高いわけではないのでここに追加するべきではないと思います
また、roleIdが指定されていないためこのタブを開くとエラーが発生します

tacko-o replied:
おっしゃる通り該当箇所を削除しようと思います
なるほど自動で追加されるタブ了解です
たしかにroleIdも指定できていなかったですね…

tacko-o committed:
ロールタイムライン取得でMisskeyRolesを使用するよう統一

https://github.com/shiosyakeyakini-info/misskey_dart/pull/24/commits/9bdce1d6e81676b32bcb8f6898b2236e36b49d18 の修正対応

tacko-o committed:
初期追加タブからロールタイムラインを削除

https://github.com/shiosyakeyakini-info/miria/pull/328#discussion_r1330747477

tacko-o committed:
Merge branch 'shiosyakeyakini-info:develop' into role_timeline

shiosyakeyakini-info commented:
マージしました！ありがとうございます！
